plasmapy/particles/tests/test_decorators.py

```diff
@@ -480,12 +524,14 @@ def __init__(
 @pytest.mark.parametrize(
     ("kwargs_to_particle_input", "arg"), kwargs_to_decorator_and_args
 )
-def test_particle_input_verification(kwargs_to_particle_input, arg) -> None:
+def test_particle_input_verification(
+    kwargs_to_particle_input: dict[str, Any], arg: Any
+) -> None:
     """Test the allow_custom_particles keyword argument to particle_input."""
 
     @particle_input(**kwargs_to_particle_input)
-    def f(particle: ParticleLike):
-        return particle
+    def f(particle: ParticleLike) -> Union[Particle, CustomParticle, ParticleList]:
+        return particle  # type: ignore[return-value]
```

Comment:
The `# type: ignore` comment here reflects that static type checkers like mypy are unable to tell that there is an implicit object conversion happening here, which would require actually running the code in `@particle_input` instead of statically analyzing it.  This is going to show up in a bunch of places throughout the code.  In these cases, it's probably fine to explicitly ignore the mypy error since it is something that mypy is unable to check.